stylesheet-loading-proposal.md

```diff
@@ -31,27 +29,6 @@ which [have beeen added](https://drafts.csswg.org/cssom/#add-a-css-style-sheet)
 
 Add a definition of a preload scanner. The preload scanner may parse and
 analyze the HTML document, or any external resources, in order to pre-load
-network resources. However, any external resource specified within a style
-sheet resource may only be pre-loaded if the resource is *guaranteed* to be needed to style the HTML document.
+network resources.
 
-Note: the pre-load scanner may affect the order in which resources are obtained, and may induce side-effects in cases where a server providing the resource over the network maintain states, and may be observable to Service Workers. The requirement to only load resources which are guaranteed to be needed is in order to avoid (a) excessive network resource use, and (b) unexpected side-effects due to style sheet rules which have no rendering effect.
-
-Example: if a style sheet has the rule:
-
-```
-.class-name {
-  background-image: url("https://example.com/example.png");
-}
-```
-
-then example.png would not be pre-loaded, because it is not guaranteed that
-there will be a DOM element with class name "class-name".
-
-However, in the following example, example.png could be pre-loaded if it the preload scanner could verify that the media query applied to the document.
-
-```
-@media (max-width: 600px)
-body {
-  background-image: url("https://example.com/example.png");
-}
-```
\ No newline at end of file
+Note: the pre-load scanner may affect the order in which resources are obtained, and may induce side-effects in cases where a server providing the resource over the network maintain states, and may be observable to Service Workers.
```

Comment:
The reason I had all that text above was based on an offline conversation with @igrigorik. He said that he thought
Chrome's preload scanner may actually try to evaluate some media queries to find more resources within
external style sheets. Was that wrong?

I felt that it was good to specify that preload scanners shouldn't load things that aren't "needed for HTML rendering". This also addresses a concern raised in the intent-to-ship thread regarding preload scanning of resources specified in CSS rules, where esprehn mentioned that one reason Chrome doesn't do this is because it would lead to too much network traffic. It seems good to specify that browsers should not do this.

Also, it seems you are saying that no preload scanners currently examine contents of external CSS.

Also, a nit, which fixes a grammar error in my original commit. "over the network maintains state"

Reply:
It only does that for responsive images in the picture element.  I'd be good with just saying that the "HTML preload scanner can scan ahead in the document for resources that are likely to be needed" but leave the heuristics for what is likely to be needed to the implementation.  The bit about "resource is guaranteed to be needed" and the loading of resources within a stylesheet felt too prescriptive.

Reply:
And correct.  AFAIK, there is only a "HTML Preload scanner" that scans ahead in the document for when the parser is blocked.  Resources in stylesheets are only obtained after styles are applied.

Reply:
> HTML preload scanner can scan ahead in the document for resources that are likely to be needed

Ok, can you add that to the PR then?